### PR TITLE
Loading data architecture docs fixes

### LINKED ISSUE #388: Mislabeled architecture doc section
Section

Architecture Guide> "Virtual Products"

Is not about "virtual products", it describes storage, and should be integrated into storage section of the "Architecture Guide"

This file should be renamed:

https://github.com/opendatacube/datacube-core/blob/develop/docs/architecture/virtual.md

or merged into 

https://github.com/opendatacube/datacube-core/blob/develop/docs/architecture/storage.rst

### PR DESCRIPTION
**Pull request by omad**
### Reason for this pull request
I didn't finish the job of incorporating @Kirill888's awesome documentation on the
data load process in Data Cube, and mangled the name to boot.

This PR finishes the job.

### Proposed changes
- Rename Virtual... -> Data Loading
- Copy edit the Data Loading documentation
- Add lots of cross references to python classes and methods
- Remove the now superfluous Data Load docs
- Add support for including MathJax equations in the docs

 - [x] Closes #388
 - ~[ ] Tests added / passed~

### CODE REVIEW

File: docs/architecture/data_load.rst [hunk 1]
```diff
@@ -1,50 +0,0 @@
...
--------------
-The basics of the datacube load function are:
-
-* `Find Datasets`_ - Look up the index​
-* `Group Datasets`_ - Sort and group by time​ (and/or other dimensions)
-* `Load Data`_ - Read from storage, write to output array
-
-Find Datasets
-~~~~~~~~~~~~~
-Base on the user query, look up the index to find a list of `dataset` objects.
```
Comment (Kirill888) [inline docs/architecture/data_load.rst:13]: `Based on`

File: docs/architecture/data_load.rst [hunk 2]
```diff
@@ -1,50 +0,0 @@
...
-but the design allows for extension to arbitrary dimensions.)​
-
-This can be the time stamp of the dataset, or a `solar_day` aggregation that group together based on time and
-longitude, to avoid splitting days at UTC midnight.​
-
-Load Data
-~~~~~~~~~
-For each group we created in the last step:​
-    For each measurement (band):​
-        For each dataset in the final chunk​
```
Comment (Kirill888) [inline docs/architecture/data_load.rst:31]: I'm pretty sure it goes over bands first, hence the problems raised in this issue #383

File: docs/architecture/data_load.rst [hunk 3]
```diff
@@ -1,50 +0,0 @@
...
-            Open the file for the dataset​
-            Read the data for the requested area​
-            Copy it to the output array in memory​
-
-Load Data Lazily (optional)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Instead of actually returning the data, we can just return the set of functions
-that would return each portion of thedata as a computation graph.​
```
Comment (Kirill888) [inline docs/architecture/data_load.rst:39]: `thedata`

File: docs/architecture/data_load.rst [hunk 4]
```diff
@@ -1,50 +0,0 @@
...
-We can then chain more operations on computation graph, and then execute across
-multiple processes on the computer, or across many computers.
-
-
-
-GridWorkflow
-------------
-The GridWorkflow class allows for dividing a query along a spatial grid, allowing the data to be split.
-
-Typically an application will match the output grid to the input grid,
```
Comment (Kirill888) [inline docs/architecture/data_load.rst:50]: `,` -> `.`

**Review by Kirill888 — CHANGES_REQUESTED** [review]
Thanks Damien, just a few minor things I noticed.

**Comment by omad** [comment]
Aaah, thanks for all the suggested changes on the file I deleted... :)

**Comment by Kirill888** [comment]
duh

**Review by Kirill888 — DISMISSED** [review]

File: docs/about/release_process.rst [hunk 5]
```diff
@@ -58,27 +58,3 @@ stable version.
...
-
-#. Notify the community of the release using the Datacube Central mailing list
-    The notifications are sent out using MailChimp_. You might need an invitation from the Geoscience
-    Australia account.
-    
-    Create a campaign (possibly by replicating one of the existing ones) and change the notes.
-    You could send out a test mail to selected accounts before sending it out to the entire DEA
-    Beta Users list.
-
-.. _PyPI: https://pypi.python.org/pypi
```
Comment (andrewdhicks) [inline docs/about/release_process.rst]: Keep these links, except for `MailChimp` and `Datacube Environment`

**Review by andrewdhicks — CHANGES_REQUESTED** [review]

**Review by andrewdhicks — DISMISSED** [review]

**Comment by omad** [comment]
*bump*

**Review by Kirill888 — APPROVED** [review]
👍

**Pull request merged**